src/plugins/vis_type_timeseries/server/lib/timeseries_visualization_client_wrapper.test.ts

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import _ from 'lodash';
+
+import { SavedObjectsClientWrapperOptions } from '../../../../core/server';
+import testParams from './test_utils/test_params.json';
+import { timeSeriesVisualizationClientWrapper } from './timeseries_visualization_client_wrapper';
+import { savedObjectsClientMock } from '../../../../core/server/mocks';
+
+jest.mock('./services', () => ({
+  getDataSourceEnabled: jest
+    .fn()
+    .mockReturnValueOnce({ enabled: false })
+    .mockReturnValue({ enabled: true }),
+}));
+
+describe('timeseriesVisualizationClientWrapper()', () => {
+  const client = savedObjectsClientMock.create();
+  client.get = jest.fn().mockImplementation((type: string, id: string) => {
+    if (type === 'data-source' && id === 'non-existent-id') {
+      return Promise.resolve({
+        id,
+        errors: {},
+      });
+    }
+    return Promise.resolve({
+      id,
+      attributes: {
+        title: `${id} DataSource`,
+      },
+    });
+  });
+  const mockedWrapperOptions = {} as SavedObjectsClientWrapperOptions;
+  mockedWrapperOptions.client = client;
+
+  const getAttributesWithParams = (params: any) => {
+    return {
+      title: 'Some TSVB Visualization',
+      visState: JSON.stringify({
+        title: 'Some TSVB Visualization',
+        type: 'metrics',
+        aggs: [],
+        params,
+      }),
+    };
+  };
+
+  beforeEach(() => {
+    client.create.mockClear();
+  });
+
+  test('if MDS is disabled, do not update the datasource references', async () => {
+    const attributes = getAttributesWithParams(testParams.withDataSourceFieldEmpty);
+    const wrapper = timeSeriesVisualizationClientWrapper(mockedWrapperOptions);
+    await wrapper.create('visualization', attributes, { references: [] });
+
+    expect(client.create).toBeCalledWith(
+      'visualization',
+      attributes,
+      expect.objectContaining({ references: [] })
+    );
+  });
+
+  test('non-visualization saved object should pass through', async () => {
+    const attributes = {
+      title: 'some-dashboard',
+    };
+    const wrapper = timeSeriesVisualizationClientWrapper(mockedWrapperOptions);
+    await wrapper.create('dashboard', attributes, { references: [] });
+
+    expect(client.create).toBeCalledWith(
+      'dashboard',
+      attributes,
+      expect.objectContaining({ references: [] })
+    );
```

Comment:
This block seems to be repeated. Can we extract this to function for better maintenance?